### PR TITLE
feat(alert): small alerts token migration and update shim

### PR DESCRIPTION
**Pull request by valentin-mladenov**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/angular/src/emphasis/alert/STYLES.md [hunk 1]
```diff
@@ -108,6 +108,12 @@
 | --clr-alert-sm-line-height                       | Alert small line height                                           |
 | --clr-alert-app-padding                          | App level alert padding                                           |
 | --clr-alert-items-gap                            | Gap between items in the alert                                    |
+| --clr-alert-sm-line-height                       | Small alert line height                                           |
+| --clr-alert-sm-font-size                         | Small alert font size                                             |
+| --clr-alert-sm-letter-spacing                    | Small alert letter spacing                                        |
+| --clr-alert-sm-font-weight                       | Small alert font weight                                           |
+| --clr-alert-sm-icon-size                         | Small alert icon size                                             |
+| --clr-alert-sm-item-text-align-items             | Small alert text align items                                      |
```
Comment (dtsanevmw) [inline projects/angular/src/emphasis/alert/STYLES.md:116]: We don't need to write down tokens which are temporary `ATM` and are about to be removed.

File: projects/angular/src/emphasis/alert/_alert.clarity.scss [hunk 2]
```diff
@@ -448,63 +448,149 @@
...
     .alert-icon-wrapper {
       padding-top: 0;
-      height: variables.$clr_baselineRem_0_667;
+      @include mixins.css-var(flex-basis, clr-alert-sm-icon-size, 1.25rem, variables.$clr-use-custom-properties);
+      @include mixins.css-var(
+        height,
+        clr-alert-sm-icon-wrapper-height,
+        variables.$clr_baselineRem_0_667,
+        variables.$clr-use-custom-properties
+      );
```
Comment (dtsanevmw) [inline projects/angular/src/emphasis/alert/_alert.clarity.scss:523]: The wrapper gets displaced from the icon inside:
Screenshot 2023-08-21 at 20 28 00
We can add `display: flex, align-self: center` to align the wrapper vertically to the center

**Review by dtsanevmw — CHANGES_REQUESTED** [review]
Check comments

File: projects/ui/src/shim.cds-core.scss [hunk 3]
```diff
@@ -457,18 +457,32 @@
...
+  // Alert small
+  --clr-alert-sm-line-height: var(--clr-p2-line-height);
+  --clr-alert-sm-font-size: var(--cds-global-typography-font-size-3);
+  --clr-alert-sm-letter-spacing: var(--clr-p2-letter-spacing);
+  --clr-alert-sm-font-weight: var(--cds-global-typography-font-weight-regular);
+  --clr-alert-sm-icon-size: var(--cds-global-space-7);
+  --clr-alert-sm-icon-wrapper-height: var(--cds-global-space-7);
+  --clr-alert-sm-padding: var(--cds-global-space-3) var(--cds-global-space-6) var(--cds-global-space-3) var(--cds-global-space-5);
+  --clr-alert-sm-item-text-align-items: center;
+  --clr-alert-sm-close-icon-height: var(--cds-global-space-10); // ATM: delete after token migration
```
Comment (williamernest) [inline projects/ui/src/shim.cds-core.scss:481]: This looks odd when selecting the button. The height/width of the button should be the same (24px), and we should add a token to increase the margin-top to get the correct vertical alignment.

Comment (valentin-mladenov) [reply projects/ui/src/shim.cds-core.scss]: Is it ok like that?
image

This change is added in commit 3b37d84

File: projects/angular/src/emphasis/alert/_alert.clarity.scss (hunk 2 above)
Comment (valentin-mladenov) [reply projects/angular/src/emphasis/alert/_alert.clarity.scss]: This change is added in commit 3b37d84

File: projects/angular/src/emphasis/alert/STYLES.md (hunk 1 above)
Comment (valentin-mladenov) [reply projects/angular/src/emphasis/alert/STYLES.md]: `clr-alert-sm-item-text-align-items` is removed in commit 3b37d84

**Review by dtsanevmw — APPROVED** [review]
Check comments

**Review by williamernest — APPROVED** [review]

**Pull request merged**